### PR TITLE
CONTRIBUTING.md improvements for commits/maintainers

### PR DESCRIPTION
**Pull request by n0toose**
<details>
<summary>Checklist</summary>

- [x] I have signed off my commits using `-s` or `Signed-off-by`\* (See: [Contributing § DCO](https://github.com/tenacityteam/tenacity/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))
- [ ] I made sure the code compiles on my machine (N/A)
- [x] I made sure there are no unnecessary changes in the code\*
- [x] I made sure the title of the PR reflects the core meaning of the issue you are solving\*
- [x] I made sure the commit message(s) contain a description and answer the question "Why do those changes fix that particular issue?" or "Why are those changes really necessary as improvements?"\*

\* indicates required

</details>

### CODE REVIEW

**Comment by n0toose** [comment]
These changes were heavily inspired by recent conversations with @emabrey and @Be-ing, as well as recent modifications that I made to commit messages for consistency.

**Review by leio — DISMISSED** [review]
My main concern in all this reverting business is that the PR whose changes got reverted remains closed afterwards. Does GH allow to reopen it and we should put that into the process , or should we put a "file open issue about getting that change back in if applicable after changes" into the process or something?

**Comment by n0toose** [comment]
> My main concern in all this reverting business is that the PR whose changes got reverted remains closed afterwards. Does GH allow to reopen it and we should put that into the process , or should we put a "file open issue about getting that change back in if applicable after changes" into the process or something?

They *cannot* be reopened, only new ones can be created - I'll try to file a ticket with GitHub in case they'll add that as a feature. **EDIT:** Done.

**Review by cklim074 — APPROVED** [review]

**Review by cklim074 — DISMISSED** [review]

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -230,3 +237,54 @@ important:
   the #nnn format, GitHub will still show the #nnn format on the website, but
   other websites and/or the command line will show the full hyperlink.
   This is good, because it reduces our dependency on GitHub.
+
+###### Proposed changes
```
Comment (Be-ing) [inline CONTRIBUTING.md]:
```suggestion
###### Merging branches
```

**Review by Be-ing — APPROVED** [review]
This doesn't address the main issues we've been having of determining when it is okay to merge, but that can be done in another PR.

**Comment by n0toose** [comment]
> This doesn't address the main issues we've been having of determining when it is okay to merge, but that can be done in another PR.

What am I missing here? (This won't affect me merging this pull request, but may be useful as a point of reference later down the line.)

**Pull request merged**